src/StateService/StatePlugin.P2P.cs

```diff
@@ -0,0 +1,43 @@
+using Akka.Actor;
+using Neo.IO.Caching;
+using Neo.Ledger;
+using Neo.Network.P2P;
+using Neo.Network.P2P.Payloads;
+using Neo.Plugins.StateService.Storage;
+using System;
+using Neo.IO;
+
+namespace Neo.Plugins.StateService
+{
+    public partial class StatePlugin : Plugin, IP2PPlugin
+    {
+        internal const string StatePayloadCategory = "StateService";
+        private readonly HashSetCache<UInt256> knownHashes = new HashSetCache<UInt256>(Blockchain.Singleton.MemPool.Capacity * 2 / 5);
+
+        bool IP2PPlugin.OnP2PMessage(Message message)
```

Comment:
If you use `OnP2PMessage`, you have to verify the message again. I think you can listen to the `Blockchain.RelayResult` event.

Here is an example:

https://github.com/neo-project/neo-modules/blob/27296b30a280139754c3cb800888ec5ab67e69f2/src/dBFT/ConsensusService.cs#L60

https://github.com/neo-project/neo-modules/blob/27296b30a280139754c3cb800888ec5ab67e69f2/src/dBFT/ConsensusService.cs#L529-L532

Reply:
Then you don't need to implement the `IP2PPlugin` interface.

Reply:
Done.